frontend/src/pages/data/DatasetView.tsx

```diff
@@ -338,7 +339,13 @@ const DatasetView = observer((props: DatasetViewProps) => {
                                 alert("Subject name cannot be empty");
                                 return;
                             }
+                            const exists = datasetContents.contents.some(obj => obj.name === subjectName);
+                            if (exists) {
+                                alert("Subject name already exists");
```

Comment:
Will this message get shown to users on the frontend? If so, maybe add a bit more detail, e.g., "Subject name already exists! Please choose a new, unique subject name."

Reply:
Sounds better! I also changed the "Subject name cannot be empty" message.